src/server/systems/ServerGameController.cpp

```diff
@@ -2,15 +2,107 @@
 
 ServerGameController::ServerGameController(LevelKey level_key) :
     GameController(level_key), m_networkingServer(new NetworkingServer()) {
+    // termbox setup
+    int ret = tb_init();
+    if (ret) {
+        fprintf(stderr, "tb_init() failed with error code %d\n", ret);
+    }
+
+    m_termboxPlayerTable.addColumn("Player ID");
+    m_termboxPlayerTable.addColumn("UPS");
+    m_termboxPlayerTable.addColumn("Avg Drift");
+    m_termboxPlayerTable.addColumn("Direction");
+    m_termboxPlayerTable.addColumn("Joinable");
+    m_termboxServerStatsTable.addColumn("Steps per second");
+    m_termboxServerStatsTable.addColumn("Updates per second");
+    m_termboxServerStatsTable.addColumn("Broadcasts per second");
+    m_termboxTextOutputArea.addLine("Ungroup Game Server Output");
+    m_termboxTextOutputArea.addLine("--------------------------");
```

Comment:
We should move all of this initialization code and the stuff in draw into its own class.
The class can have `update` `preUpdate` `postUpdate` and `draw` methods.

I think it's really important to keep the core game loop as simple and understandable as possible.



This is how it looks on the client side:

**init**
https://github.com/SourenP/ungroup_game/blob/b47a337b15ba2544f5986de6805269b170fe6bfa/src/client/systems/ClientGameController.cpp#L48-L49
**draw**
https://github.com/SourenP/ungroup_game/blob/b47a337b15ba2544f5986de6805269b170fe6bfa/src/client/systems/ClientGameController.cpp#L57-L59